src/renderer/atlas/AtlasEngine.h

```diff
@@ -288,6 +288,12 @@ namespace Microsoft::Console::Render
             }
 #pragma warning(pop)
 
+            void destroy() noexcept
+            {
+                std::destroy_n(_data, _size);
+                deallocate(_data);
+            }

```

Comment:
These 3 `Buffer` changes should've been here all along. The 3 corresponding STL functions already do exactly the optimal thing: nothing for trivial types (i.e. leave the memory uninitialized) and properly construct/destruct the types if they're proper classes. And in case of `std::uninitialized_copy_n` in particular it already uses `memmove` for trivial types internally. Perfect!
This change allows us to construct a `Buffer<iterator>` which is an array of non-trivial types.

BTW/FYI: I didn't use `std::vector` and wrote `Buffer` instead, because I want to convey that these things explicitly _don't_ hold resizeable contents (in general). For instance I didn't want a `push_back` method on my viewport arrays - that wouldn't make sense so that always felt a bit wrong to me. It also doesn't initialize contents to zero, allowing rapid creation/destruction and you can specify a custom alignment (improves rendering perf by about 20% over `std::vector` IIRC).

Reply:
When you pull these classes out to their own files, explain that ^^ in a comment!